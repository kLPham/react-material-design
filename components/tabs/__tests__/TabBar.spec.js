import React from 'react';
import { shallow } from 'enzyme';
import TabsBar from '../index';
import Tab from '../tab';

describe('<TabsBar />', () => {
    it('should render as basic', () => {
        const tabBar = shallow(
            <TabsBar>
                <Tab active label="Home" />
                <Tab label="Marketing" />
            </TabsBar>,
        );
        expect(tabBar).toMatchSnapshot();
    });
    it('should render with icons', () => {
        const tabBar = shallow(
            <TabsBar>
                <Tab active icon="home" />
                <Tab icon="phone" />
            </TabsBar>,
        );
        expect(tabBar).toMatchSnapshot();
    });
    it('should render with icons and text', () => {
        const tabBar = shallow(
            <TabsBar>
                <Tab active label="Home" icon="home" />
                <Tab label="Marketing" icon="phone" />
            </TabsBar>,
        );
        expect(tabBar).toMatchSnapshot();
    });
});
