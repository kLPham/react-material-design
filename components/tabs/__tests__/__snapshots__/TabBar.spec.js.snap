// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabsBar /> should render as basic 1`] = `
<nav
  className="basic-tab-bar"
>
  <Tab
    active={true}
    label="Home"
  />
  <Tab
    label="Marketing"
  />
  <span
    className="mdc-tab-bar__indicator"
  />
</nav>
`;

exports[`<TabsBar /> should render with icons 1`] = `
<nav
  className="mdc-tab-bar mdc-tab-bar--icon-tab-bar"
>
  <Tab
    active={true}
    icon="home"
  />
  <Tab
    icon="phone"
  />
  <span
    className="mdc-tab-bar__indicator"
  />
</nav>
`;

exports[`<TabsBar /> should render with icons and text 1`] = `
<nav
  className="mdc-tab-bar mdc-tab-bar--icons-with-text"
>
  <Tab
    active={true}
    icon="home"
    label="Home"
  />
  <Tab
    icon="phone"
    label="Marketing"
  />
  <span
    className="mdc-tab-bar__indicator"
  />
</nav>
`;
